File: 5_Association Rule Learning/Apriori.R
# Importing dataset
#dataset = read.csv('Market_Basket_Optimisation.csv', header = FALSE)

#install.packages('arules')
library(arules)
dataset = read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE)
summary(dataset)

itemFrequencyPlot(dataset, topenv = 100)

# Training apriori on the dataset
total_rows = nrow(dataset)
#Support a product that purchased 3 times a day for over the whole week
# 3x7/total_rows = 0.003
rules = apriori(data = dataset, parameter = list(support = 0.003, confidence = 0.2))

# Visualising the results
inspect(sort(rules, by = 'lift')[1:10])

#Support a product that purchased 4 times a day for over the whole week
# 3x7/total_rows = 0.003
rules = apriori(data = dataset, parameter = list(support = 0.004, confidence = 0.2))

# Visualising the results
inspect(sort(rules, by = 'lift')[1:10])

File: 5_Association Rule Learning/Apriori.py
# Importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing data
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header=None)

transactions = []
for i in range(0, len(dataset)):
    transactions.append([str(dataset.values[i, j]) for j in range(0, len(dataset.columns))])

# Training apriori on the dataset
from apyori import apriori
min_support = 3*7/len(dataset)
rules = apriori(transactions, 
                min_support = 0.003,
                min_confidence=0.2,
                min_lift = 3,
                min_length=2)

# Visualising the results
results = list(rules)